run.cc

```diff
@@ -60,20 +61,84 @@ void runTests() {
     std::cout << (test() ? " ok" : " FAILED!") << std::endl;
 }
 
+
 int main() {
   using namespace std;
+  // Exercise 1c)
+  std::vector<Data> allSets;
+  std::string filenames[] = {"exp_A", "exp_B", "exp_C", "exp_D"};
 
   cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1a)" << endl;
   runTests();
   cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1b)" << endl;
   // create an object which holds data of experiment A
   Data datA("exp_A");
-
   // here is the data from experiment A
   cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
        << endl;
   cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
        << endl;
+  // Exercise 1b)
+  cout << "uncertainty of experiment A in bin 27: " << datA.error(27)
+       << endl << endl;
+
+  // Exercise 1c)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1c)" << endl;
+  for (int i = 0; i < 4; i++) {
+    Data set(filenames[i]);
+    allSets.push_back(set);
+    cout << "cros-section of experiment " << filenames[i]
+         << " in bin 27: " << set.measurement(27)
+         << " +- " << set.error(27) << endl;
+  }
+  
+  // Exercise 1d), 2a)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1d), 2a)" << endl;
+  int d;
+  for (int i = 0; i <=2; i++) {
+    for (int j = i + 1; j <= 3; j++) {
+      cout << "Compatibility check of " << filenames[i] 
+           << " with " << filenames[j] << ", measurements differing"
+           << endl;
+      for (int n = 1; n <= 3; n++) {
+        d = allSets[i].checkCompatibility(allSets[j], n);
+        cout << "    more than " << n << " * sigma = " << d << " points" << endl;
+      }
+      cout << endl;
+    }
+  }
+
+  // Exercise 1e)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1e)" << endl;
+  Data sum12 = allSets[0].average(allSets[1]);
+  cout << "cros-section of experiment A & B"
+       << " in bin 27: " << sum12.measurement(27)
+       << " +- " << sum12.error(27) << endl;
+
+  // Exercise 2b)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 2b)" << endl;
+  int ndf = 52;
+  for (int i = 0; i < 4; i++) {
+    cout << "X^2 / ndf for experiment " << filenames[i]
+         << " = " << allSets[i].chi2() / ndf  << endl;
+ }
+
```

Comment:
You didn't answer if the data are compatible with the model using the given parameters. The chisquare values you printed out are correct, but do you think they are 'good'? What should the value ideally be and are the deviations from that value that you see here significant? -0.5p